### ISSUE TITLE
References to `<username>@users.noreply.github.com` should be reworked, as they are misleading

### ISSUE THREAD

**Issue by jsoref** [github/docs#27985]
👉 To the contributor who picks up this issue: See the **content plan to follow** below, under the header "Maintainer note."

### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/adding-content-to-your-github-pages-site-using-jekyll

### What part(s) of the article would you like to see updated?

There are several references to `<username>@users.noreply.github.com` email addresses in the docs:

https://github.com/search?q=repo%3Agithub%2Fdocs+%3Cusername%3E%40&type=code

This is slightly misleading. The correct behaviour is described in a note on "[Setting your commit email address](https://github.com/github/docs/blob/main/content/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address.md?plain=1#L52)":

> Note: If you created your account on GitHub.com after July 18, 2017, your noreply email address for GitHub is an ID number and your username in the form of [ID+USERNAME@users.noreply.github.com](mailto:ID+USERNAME@users.noreply.github.com). If you created your account on GitHub.com prior to July 18, 2017, and enabled Keep my email address private prior to that date, your noreply email address from GitHub is [USERNAME@users.noreply.github.com](mailto:USERNAME@users.noreply.github.com). You can get an ID-based noreply email address for GitHub by selecting (or deselecting and reselecting) Keep my email address private in your email settings.

### Additional information

I made the mistake of relying on this (or equivalent) documentation and then people told me I was doing it wrong...

You can see all the times I did it wrong in:
https://github.com/search?q=%22jsoref%40users.noreply.github.com%22+%2F%5B%5E%2B%5Djsoref%40users.noreply.github.com%2F&type=code
And you can see that I'm not the only one who made that mistake if you adjust the query a bit.

You can see more recent things where I'm doing it "right":
https://github.com/search?q=%222119212%2Bjsoref%40users.noreply.github.com%22&type=pullrequests
https://github.com/search?q=%222119212%2Bjsoref%40users.noreply.github.com%22&type=code

Here's a commit
https://github.com/check-spelling/check-spelling-docs/commit/0643ed1315fc1df51858307573a4f123d494df29.patch generated by using the web editor -- https://github.com/check-spelling/check-spelling/wiki/_compare/86e0773ecf7ad1c138ee8a01a9499118ec7301a9...0643ed1315fc1df51858307573a4f123d494df29

### Maintainer note

Where references to `<username>@users.noreply.github.com` exist (see [the search](https://github.com/search?q=repo%3Agithub%2Fdocs+%3Cusername%3E%40&type=code)), we should change this language to be more general about a no-reply email address that will vary depending on when the username was changed, then link to the article https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address. 

For example:

> ... the commit author email address cannot be changed and is `<username>@users.noreply.github.com` by default

Might become:

> ... the commit author email address cannot be changed and will be a no-reply by default. For more information about the exact form the no-reply email address can take, see "[AUTOTITLE](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address)."

Where AUTOTITLE is `[AUTOTITLE](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address)` in the Markdown.

**Comment by ShatilKhan** [comment]
Hey, I can take a look at it. If possible assign it to me please.

**Comment by jsoref** [comment]
Just create a PR and include a `fixes #... ` in the PR.

**Comment by DARKo6o6o6** [comment]
I have no clue what I am doing

**Comment by cmwilson21** [comment]
@jsoref Thanks for opening an issue. I'll get it triaged for review 👍

**Comment by subatoi** [comment]
Thanks @jsoref—I'm not sure this is quite as straightforward as it being wrong, though. Or at least not completely. There was a change back in 2017: https://github.blog/2017-04-11-private-emails-now-more-private/, so for _some_ users it's correct.

I see your account was making contributions before 2017, so perhaps you could confirm if what's in the following screenshot makes sense (in your case, the part about "If you created your account on GitHub.com prior to July 18, 2017, and enabled **Keep my email address private prior to that date**"): 

<img width="665" alt="Screenshot 2023-09-07 at 08 53 43" src="https://github.com/github/docs/assets/32935794/ec0d47af-9066-436b-b52f-fce1aade7316">

For reference, that's from https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address

If so, I'd suggest the best way to resolve this would be to convert that into a reusable and replace the existing incomplete/misleading text.

**Comment by jsoref** [comment]
Yes, that seems likely. Although at this point, I'm getting the longer form.

**Comment by subatoi** [comment]
Sounds good 👍: in that case I think the best way to resolve it is to convert the aforementioned note into a reusable and replace the existing incomplete/misleading text with it. 

If you or anyone who sees that is interested in taking this on, please let me know (I'm happy to clarify what I mean if anything's not clear). Else, I'll make this change myself :) 

Thanks for raising this!

**Comment by subatoi** [comment]
@jsoref I took another look at this in advance of Hacktoberfest, and in hindsight it probably doesn't need to be a reusable, and we can instead correct the misleading references to be more generic and link to https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address. 

I amended the issue description a little to add detail about the content changes needed.

Let me know what you think as there are different ways to resolve this. If you're happy with this as is, I'll add the `help-wanted` label so we can get this ready for Hacktoberfest 🎃
